components/raftstore-v2/src/operation/ready/mod.rs

```diff
@@ -527,10 +557,27 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
                 .flat_map(|m| self.build_raft_message(m))
                 .collect();
         }
+        if self.has_pending_messages() {
+            if write_task.messages.is_empty() {
```

Comment:
Why check?

Reply:
Avoid unnecessary allocation and memcopy.